pkg/deploy/controller/image_change_controller.go

```diff
@@ -80,6 +82,34 @@ func (c *ImageChangeController) HandleImageRepo() {
 	}
 }
 
+// triggerMatchesImages decides whether a given trigger for config matches the provided image repo.
+// When matching:
+// * The trigger From field is preferred over the deprecated RepositoryName field.
+// * The namespace of the trigger is preferred, over the config namespace.
+// * The repo Status.DockerImageRepository field is preferred over the repo DockerImageRepository field.
+func triggerMatchesImage(config *deployapi.DeploymentConfig, trigger *deployapi.DeploymentTriggerImageChangeParams, repo *imageapi.ImageRepository) bool {
+	if len(trigger.From.Name) > 0 {
+		namespace := trigger.From.Namespace
+		if len(namespace) == 0 {
+			namespace = config.Namespace
+		}
+
+		if repo.Namespace == namespace && repo.Name == trigger.From.Name {
+			return true
+		}
+	}
+
+	if len(trigger.RepositoryName) > 0 {
+		if trigger.RepositoryName == repo.Status.DockerImageRepository {
```

Comment:
With that change, you can invert `if len(trigger.RepositoryName) > 0` to have it return false when the repository name is empty